### ISSUE TITLE
Opensearch internal users do not persist

### ISSUE THREAD

**Issue by rauldpm** [wazuh/wazuh-docker#638]
Version: v4.3.0
Deployment: single-node

Hi team

It has been found that currently creating a user in `OpenSearch Plugin -> Security -> Internal Users` does not persist when stopping or restarting the container with `docker-compose stop/restart`

The user also doesn't persist creating it with the API call:

- curl -k -X PUT -u admin:SecretPassword https://IP:9200/_opendistro/_security/api/internalusers/tested -H "Content-Type: application/json"  --data '{  "password": "Mypassword1@" }'

Users can be verified by calling the API:

- curl -k -XGET -u admin:SecretPassword https://IP:9200/_opendistro/_security/api/internalusers/?pretty

This should be investigated since even using local mount points of the indexes, which are located in the folder (/var/lib/wazuh-indexer), there seems to be no persistence of them.

This behavior has also been found in 4.2.6

- Additionally, it may be interesting mount /usr/share/wazuh-indexer/config in a volume to allow config data persist.

Regards, Raúl.

**Comment by rauldpm** [comment]
### Additional data

- Indices api call

```
# curl -k -XGET -u admin:SecretPassword https://192.168.57.123:9200/_cat/indices
green open wazuh-statistics-2022.19w   ce-it3c2RW-DEI778vtm2A 1 0 43 0 241.9kb 241.9kb
green open wazuh-alerts-4.x-2022.05.13 Xi7_3lutRvCuF1MVOL8-yw 3 0  8 0  92.9kb  92.9kb
green open wazuh-alerts-4.x-2022.05.12 7j-2WTDERX-1fIWKOdbEWw 3 0  1 0  12.1kb  12.1kb
green open .opendistro_security        ah_JbAlAQEOqnUOvr_c9gQ 1 0  9 4 125.4kb 125.4kb
green open .kibana_1                   2nt22KiITrSC4CGecHwSHA 1 0  4 0  14.7kb  14.7kb
green open wazuh-monitoring-2022.19w   hexWyyphTk6jPTc__RENqw 1 0  0 0    208b    208b
```

- Mounted indices in host system

```
# ls -l /wazuh/wazuh-indexer-data/nodes/0/indices/
total 24
drwxrwxr-x 4 vagrant vagrant 4096 May 13 16:23 2nt22KiITrSC4CGecHwSHA
drwxrwxr-x 6 vagrant vagrant 4096 May 13 16:23 7j-2WTDERX-1fIWKOdbEWw
drwxrwxr-x 4 vagrant vagrant 4096 May 13 16:23 ah_JbAlAQEOqnUOvr_c9gQ
drwxrwxr-x 4 vagrant vagrant 4096 May 13 16:23 ce-it3c2RW-DEI778vtm2A
drwxrwxr-x 4 vagrant vagrant 4096 May 13 16:23 hexWyyphTk6jPTc__RENqw
drwxrwxr-x 6 vagrant vagrant 4096 May 13 16:23 Xi7_3lutRvCuF1MVOL8-yw
```

**Comment by rauldpm** [comment]
### Update report

After commenting the problem with @vcerenu , it has been concluded that the cause of this is the execution of the securityadmin.sh script every time the container is started, restarting the saved data.

https://github.com/wazuh/wazuh-docker/blob/73018f87ac49441c4d5695cc8db486a33ff195c1/build-docker-images/wazuh-indexer/config/entrypoint.sh#L89

This script should be executed only once during the creation of the container, if it is necessary to execute it more times, it will be because the user has modified something and it is necessary, but not as normal. A possible solution would be to use a flag for the entrypoint to detect if the cluster has been initialized, although this would have to be done by perhaps making a call and checking the status.

```
if [[ "$DISCOVERY" == "single-node" ]] && [[ ! -f "/usr/share/wazuh-indexer/config/.flag" ]]; then
  touch "/usr/share/wazuh-indexer/config/.flag"
  # run securityadmin.sh for single node with CACERT, CERT and KEY parameter
  nohup /securityadmin.sh &
fi
```

**Comment by teddytpc1** [comment]
This issue was corrected on `v4.3.1` with this [commit](https://github.com/wazuh/wazuh-docker/commit/65fd592d52e60db667493354adbae0fce19385a9).
